Add site settings blogger importer

talldan opened the pull request:
Adds a new importer for blogger.com to the list of importers in Settings > Import.

The blogger importer works similarly to other importers. Users can export an xml file from their blogger.com site, and upload that file to WordPress using the importer. The importer will extract posts, pages, comments, images and will convert blogger.com labels to WordPress tags (the conversion process is handled by the backend).

This task is part of a trial project.

### Test Plan ###
**Prerequisites**
In order to test the import process on a sandbox environment, you'll need to run some asynchronous jobs and setup logging to a Jabber client. Follow these steps:
1. On your sandbox, edit the `public_html/wp-content/admin-plugins/restapi-import-manager.php` file, find the function `should_log_import`, and add your wp username to the `$debug_users` array.
2. Follow the instructions in the field guide `PCYsg-5Xx-p2` to setup Jabber logging (if you haven't already)

**Testing**
1. Export data from a blogger.com site (Settings > Other > Back up content)
2. In WordPress visit the Imports page (Settings > Import)
3. Using the Blogger.com importer upload the exported blogger.com file
4. Follow the steps to complete the import, and confirm that the import finishes successfully

**Completing testing on a sandbox**
If using a sandbox to test, you'll have to finish the imports manually by running an async job. Make sure you've followed the steps in the 'prerequisites' section. Then:
1. When starting the import, view your jabber debug log, you should see some output during the import process. Find the lines that looks:
```
[    DEBUG] Queued import session 1a2b3c4d5e6f7
[    DEBUG]  - job id: 12345678901
```
2. Ssh into your sandbox and cd to the `public_html/async-jobs/` folder.
3. Run the command `./tester.php 12345678901` using the job id from your jabber log
4. You should see some output in your console and some further output in your jabber client, if the command ran successfully
5. In the jabber client find the lines that look like:
```
[    DEBUG]  - executing import (log? = 1)
[    DEBUG]  - queued finish job 56789012345
```
6. In your terminal run the command `./tester.php 56789012345` using the job id from your jabber log
7. The command should complete successfully, and back in your browser window the import will show as completed.

<img width="400" alt="screen shot 2018-04-20 at 17 04 01" src="https://user-images.githubusercontent.com/677833/39042229-e2168bae-44bc-11e8-9f5d-8c4605a2e3b8.png">

spen commented on client/my-sites/importer/importer-blogger.jsx:
```diff
@@ -0,0 +1,47 @@
+	static displayName = 'ImporterBlogger';
+
+	render() {
+		importerData.description = this.props.translate(
```
We can improve on this pattern to avoid the slight misdirection that comes from having `importerData` up top and adding to it here...

See https://github.com/Automattic/wp-calypso/pull/23930#discussion_r179598308

spen commented on client/my-sites/importer/importer-blogger.jsx:
```diff
@@ -0,0 +1,47 @@
+			'Upload a {{b}}Blogger.com export file{{/b}} to start ' +
+				'importing into {{b2}}%(title)s{{/b2}}.',
+			{
+				args: { title: this.props.site.title },
```
We should add `site` as a required prop using `propTypes` (making use of `shape` to ensure it has a `title` too)

talldan replied:
Thanks - yep, agree with the comments. I'll make sure to replicate the improvements to importer-medium.jsx, which I took a lot of the code from.

I did find missing propTypes a bit confusing, so will make sure to add them!

spen replied:
Thanks @talldan :)

My rule of thumb is if it's chained (and thus could through a missing ref error) it's worth either being mentioned in propTypes or using `get` instead to safely grab the chained value :)

spen commented on client/my-sites/importer/importer-blogger.jsx:
```diff
@@ -0,0 +1,47 @@
+				args: { title: this.props.site.title },
+				components: {
+					b: <strong />,
+					b2: <strong />,
```
Is there significance in the second `strong` tag here? We should be able to use just the first and repeat it where needed I think.

a8ci18n commented on client/my-sites/importer/importer-medium.jsx:
```diff
@@ -12,34 +13,34 @@ import { localize } from 'i18n-calypso';
+			'Import posts, tags, images, and videos from a Medium export file.'
+		),
+		uploadDescription: this.props.translate(
+			'Upload a {{b}}Medium export file{{/b}} to start importing into {{b}}%(title)s{{/b}}.',
```
Hi! I've found a possible matching string that has already been translated **20** times:
`translate( 'Upload a {{b}}Medium export file{{/b}} to start importing into {{b2}}%(title)s{{/b2}}.' )` *ES Score: 15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

bisko commented:
I like the way you're improving the code! Do you think it would be possible to split the PR in several smaller PRs, as it will help with more focused reviews and not introducing too many new things at once?

talldan commented:
@bisko - sure thing. I've introduced:
- https://github.com/Automattic/wp-calypso/pull/24349 for the refactoring of the way importers are rendered
- https://github.com/Automattic/wp-calypso/pull/24350 for making the importer components consistent

and I've removed those commits from this PR and addressed the code review feedback in separate commits

akirk commented on client/my-sites/importer/importer-blogger.jsx:
```diff
@@ -0,0 +1,58 @@
+			'Import posts, pages, comments, tags, and images from a Blogger.com export file.'
+		),
+		uploadDescription: this.props.translate(
+			'Upload a {{b}}Blogger.com export file{{/b}} to start importing into {{b}}%(title)s{{/b}}.',
```
Would it make sense to make "Blogger.com" a variable so that the string can be reused for other importers? In the end we don't want translators to translate it anyway, right?

talldan replied:
Yep - that makes sense.

talldan replied:
I've incorporated that change into this commit - 078c1c9

akirk replied:
Thanks!

talldan commented on config/wpcalypso.json:
```diff
@@ -63,6 +63,7 @@
 		"manage/edit-user": true,
 		"manage/export/guided-transfer": true,
 		"manage/import/medium": true,
+		"manage/import/blogger": false,
```
👍 I've rebased to apply that change to this commit - 5c4db0e

akirk commented on client/my-sites/importer/importer-blogger.jsx:
```diff
@@ -0,0 +1,68 @@
+			{
+				args: {
+					importerName,
+					title: this.props.site.title,
```
One more thing, can we change the variable title to `%(siteTitle)s`. Then translators won't have to ask :) Thanks!

talldan replied:
Hi - that's now updated as well.

talldan commented:
superceded by #25312, so closing this.
